File: Data Structures & Algorithms/partition-equal-subset-sum/submission-1.java
class Solution {
    Boolean[][] dp;
    public boolean Sum(int[] nums, int idx, int sum, int target){
        if(sum==target) return true;
        if(idx==nums.length || sum>target) return false;
        if(dp[idx][sum]!=null) return dp[idx][sum];
        return dp[idx][sum]=Sum(nums, idx+1, sum+nums[idx], target) || Sum(nums, idx+1, sum, target);
    }
    public boolean canPartition(int[] nums) {
        int sum=0;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
        }
        if(sum%2!=0) return false;
        int target=sum/2;
        dp=new Boolean[nums.length][target+1];
        return Sum(nums, 0, 0, target);  
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        hash_set=set()
        nums.sort()
        for i in range(len(nums)):
            target=0-nums[i]
            j,k=i+1,len(nums)-1
            while j<k:
                if nums[j]+nums[k]==target:
                    hash_set.add(tuple([nums[i],nums[j],nums[k]]))
                    k-=1
                    j+=1
                elif nums[j]+nums[k]>target:
                    k-=1
                else:
                    j+=1
        
        return list(hash_set)